src/utils/number.ts

```diff
@@ -0,0 +1,38 @@
+export const PRECISION = 10
+
+export const EPSILON = Math.pow(10, -PRECISION - 1)
+
+export function fuzzyEquals(number1: number, number2: number) {
+  return Math.abs(number1 - number2) < EPSILON
+}
+
+export function valueInRange(value, min, max) {
+  const result = fuzzyCheckRange(value, min, max)
+  if (result !== null) return result
+}
+
+export function fuzzyCheckRange (number: number, min: number, max: number) {
+  if (fuzzyEquals(number, min)) return min
+  if (fuzzyEquals(number, max)) return max
+  if (number > min && number < max) return number
+  return null
+}
+
+export function fuzzyRound(number: number) {
```

Comment:
I think we'd better change this parameter to `num` or `target` since `number` is a keyboard in TypeScript